### PR TITLE
Split store add dynamic date range

### PR DESCRIPTION
**Pull request by Rashmil-1999**
This PR splits the Store into Filters and Data and sets the mount point of the Filters in `./Filters/index.tsx` file. Added Dynamic `minDate` and `maxDate` to the `startDate` and `endDate` date pickers. and shows error message when the user manually enters the wrong/invalid range. Also updated the Downloads button imports

### CODE REVIEW

File: src/components/Explore/Sidebar.tsx [hunk 1]
```diff
@@ -44,12 +45,18 @@ const Sidebar = () => {
                         ]}
                     />
                     <Box sx={{ alignSelf: 'center', zIndex: 1 }}>
-                        <Button
-                            variant="outlined"
-                            onClick={() => dataActionDispatcher({ type: 'updateSelectedStation', station: null })}
-                        >
-                            Go Back
-                        </Button>
+                        <Stack direction="column" spacing={4}>
```
Comment (ka7eh) [inline src/components/Explore/Sidebar.tsx:48]: Why are these needed? Looks like they can be removed?
Here's another thought: can we move the download button next to the "Go Back" button? This way it stays visible on all tabs.

File: src/components/Explore/Filters.tsx [hunk 2]
```diff
@@ -66,39 +91,55 @@ const Filters = () => {
...
+                            maxDate={maxStartDate}
+                            openTo="year"
+                            clearable
+                            onChange={(newVal) => {
+                                setStartDate(newVal);
+                            }}
+                        />
+                        <Box my={1}>
+                            <Button
+                                startIcon={<Icon baseClassName="icons">download</Icon>}
```
Comment (ka7eh) [inline src/components/Explore/Filters.tsx:109]: Looks like `download` snuck into this! This should be `close` icon! :)

**Review by ka7eh — CHANGES_REQUESTED** [review]
Looking good! I just added a few comments.

File: src/components/Explore/Sidebar.tsx (hunk 1 above)
Comment (Rashmil-1999) [reply src/components/Explore/Sidebar.tsx:48]: So the problem is I am stringifying the `stationDetails` object so on the upper component, we don't have the details yet. That is why I move it inside. one thing is we can move the state of `stationDetails` up and then add the download button there.

File: src/components/Explore/Filters.tsx (hunk 2 above)
Comment (Rashmil-1999) [reply src/components/Explore/Filters.tsx:109]: oops

Comment (Rashmil-1999) [reply src/components/Explore/Filters.tsx:109]: Also the problem here is that the close button becomes a bit weird. Will need to make the button round so as to make it look good.

**Review by ka7eh — APPROVED** [review]
